### PR TITLE
Use module-scope fixture for testing (GSI 274)

### PR DESCRIPTION
**Pull request by TheByronHimes**
Changed joint_fixture to joint_fixture_function and created a get_joint_fixture (so the same style as what is now in hexkit). 
Added `module_scope_fixtures.py` to contain the new fixtures.

The module-level fixtures greatly reduced the test duration for the edge case tests, but had little effect on the typical journey tests, so the latter remained mostly untouched.

### CODE REVIEW

File: tests/fixtures/joint.py [hunk 1]
```diff
@@ -80,3 +89,14 @@ async def joint_fixture(
...
+
+def get_joint_fixture(scope: _ScopeName = "function"):
+    """Produce a joint fixture with desired scope"""
+    return pytest_asyncio.fixture(joint_fixture_function, scope=scope)
+
+
+joint_fixture = get_joint_fixture()
+mongodb_fixture = get_mongodb_fixture()
+kafka_fixture = get_kafka_fixture()
+s3_fixture = get_s3_fixture()
```
Comment (mephenor) [inline tests/fixtures/joint.py:102]: This is not sensitive to order, i.e. joint_fixture does not need to be at the bottom?

**Review by mephenor — DISMISSED** [review]

File: tests/fixtures/joint.py (hunk 1 above)
Comment (TheByronHimes) [reply tests/fixtures/joint.py:102]: No, because it's just a function at this point. Pytest evaluates them during arrangement in the test module itself (I believe)

**Review by Cito — DISMISSED** [review]

**Review by mephenor — APPROVED** [review]
